ballista/core/src/execution_plans/shuffle_reader.rs

```diff
@@ -547,15 +581,15 @@ fn send_fetch_partitions(
     join_handles.push(tokio::spawn(async move {
         for p in remote_locations.iter() {
             let now = Instant::now();
-            let permit = response_sender.reserve().await.unwrap();
+            let permit = semaphore.clone().acquire_owned().await.unwrap();
```

Comment:
It's unlikely but we should handle the error here because technically the semaphore can be dropped before this executes

Reply:
I saw such errors in the past as in case of query timeout it gets dropped from the client side causing panic

Reply:
Updated